.gitlab/pipeline/check.yml

```diff
@@ -8,6 +8,7 @@ cargo-clippy:
     RUSTFLAGS: "-D warnings"
   script:
     - SKIP_WASM_BUILD=1 cargo clippy --all-targets --locked --workspace
+    - SKIP_WASM_BUILD=1 cargo clippy --all-targets --all-features --locked --workspace
```

Comment:
Should we have a different step for this? Or modify the one we already have?
Or does clippy do some magic and not re-do the work after the first command?

Reply:
I thought I'd add it here to hijack the previous job since it has already downloaded the packages and only has to recompile workspace packages which have other features (no other deps). You can actually see it in [the clippy job](https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4715326) that has run on this PR. Happy to make it a separate step if the CI folks are happy with the extra cost (if there is any - I don't know the ins and outs of forklift).
The features are not purely additive, and can actually remove code from the main build so I don't think we should replace the previous step.